service-base/src/main/java/org/eclipse/hono/service/metric/Metrics.java

```diff
@@ -106,11 +131,57 @@ void reportTelemetry(
             MetricsTags.TtdStatus ttdStatus,
             Sample timer);
 
+    /**
+     * Reports a telemetry message or event received from a device.
+     *
+     * @param type The type of message received, e.g. <em>telemetry</em> or <em>event</em>.
+     * @param tenantId The tenant that the device belongs to.
+     * @param tenantObject The tenant configuration object.
+     * @param outcome The outcome of processing the message.
+     * @param qos The delivery semantics used for sending the message downstream.
+     * @param payloadSize The number of bytes contained in the message's payload.
+     * @param ttdStatus The outcome of processing the TTD value contained in the message.
+     * @param timer The timer indicating the amount of time used
+     *              for processing the message.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     * @throws IllegalArgumentException if type is neither telemetry nor event or
+     *                    if payload size is negative.
+     */
+    void reportTelemetry(
+            MetricsTags.EndpointType type,
+            String tenantId,
+            TenantObject tenantObject,
+            MetricsTags.ProcessingOutcome outcome,
+            MetricsTags.QoS qos,
+            int payloadSize,
+            MetricsTags.TtdStatus ttdStatus,
+            Sample timer);
+
+    /**
+     * Reports a command &amp; control message being transferred to/from a device.
+     *
+     * @param direction The command message's direction.
+     * @param tenantId The tenant that the device belongs to.
+     * @param outcome The outcome of processing the message.
+     * @param payloadSize The number of bytes contained in the message's payload.
+     * @param timer The timer indicating the amount of time used
+     *              for processing the message.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     * @throws IllegalArgumentException if payload size is negative.
+     */
+    void reportCommand(
+            MetricsTags.Direction direction,
+            String tenantId,
+            MetricsTags.ProcessingOutcome outcome,
+            int payloadSize,
+            Sample timer);
+
     /**
      * Reports a command &amp; control message being transferred to/from a device.
      *
      * @param direction The command message's direction.
      * @param tenantId The tenant that the device belongs to.
+     * @param tenantObject The tenant configuration object.
```

Comment:
the JavaDoc needs to explain when and why I would want to use this variant of the method, wouldn't you agree?

Reply:
I will update it.